### PR TITLE
update key address

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

**Comment by disksing** [comment]
@siddontang @ngaut

File: txn.go [hunk 1]
```diff
@@ -239,10 +283,31 @@ func (txn *tikvTxn) Commit() error {
...
+	if len(prewriteReqs) == 0 {
+		return nil
+	}
+	txn.version = kv.Version{Ver: txn.tid}
+	var lastErr error
+	for _, regionID := range prewriteReqs {
+		singlePrewrite := multiPrewrite[regionID]
+		if err := txn.doCommit(singlePrewrite); err != nil {
+			log.Errorf("Commit error[%s]", err)
+			lastErr = err
```
Comment (siddontang) [inline txn.go]: no need break?

Comment (disksing) [inline txn.go]: Looks strange to me too. @morefreeze 
For now I suggest we keep it and add an issue for reviewing later again. @siddontang

Comment (siddontang) [inline txn.go]: got it

File: txn.go [hunk 2]
```diff
@@ -239,10 +283,31 @@ func (txn *tikvTxn) Commit() error {
...
+		}
+	}
+	if lastErr != nil {
+		return errors.Trace(lastErr)
+	}
+	txn.close()
```
Comment (siddontang) [inline txn.go]: need close the transaction before return error?

Comment (disksing) [inline txn.go]: I believe we should close it. In both tidb/localstore and tidb/hbase, the txn is always closed after calling Commit.

**Comment by siddontang** [comment]
LGTM
But I very recommend that we should clear and refactor this repository, add more test later.

**Pull request merged**